seacrowd/sea_datasets/cub_bahasa/cub_bahasa.py

```diff
@@ -0,0 +1,391 @@
+        195: '195.Carolina_Wren',
+        196: '196.House_Wren',
+        197: '197.Marsh_Wren',
+        198: '198.Rock_Wren',
+        199: '199.Winter_Wren',
+        200: '200.Common_Yellowthroat'
+    }
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "image_id": datasets.Value("int64"),
+                    "class_id": datasets.Value("int64"),
+                    "image_path": datasets.Value("string"),
+                    "class_name": datasets.Value("string"),
+                    "captions": [
+                        {
+                            "caption_eng": datasets.Value("string"),
+                            "caption_ind": datasets.Value("string"),
+                        }
+                    ]
+                }
+            )
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.image_text_features(label_names=list(self.IMAGE_CLASS.values()))
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        # expect several minutes to download image data ~1.2GB
+        data_path = dl_manager.download_and_extract(_URLS)
+
+        # working with image dataset
+        image_meta = Path(data_path["image"]) / "CUB_200_2011" / "images.txt"
+        df_image = pd.read_csv(image_meta, sep=" ", names=["image_id", "image_path"])
+        df_image['image_path'] = df_image['image_path'].apply(lambda x: Path(image_meta.parent, 'images', x))
+
+        label_meta = Path(data_path["image"]) / "CUB_200_2011" / "image_class_labels.txt"
+        df_label = pd.read_csv(label_meta, sep=" ", names=["image_id", "class_id"])
+
+        # working with label
+        class_meta = Path(data_path["image"]) / "CUB_200_2011" / "classes.txt"
+        self.IMAGE_CLASS = {}
+        with open(class_meta, "r") as f:
+            for line in f:
+                class_id, class_name = line.strip().split()
+                self.IMAGE_CLASS[int(class_id)] = class_name
```

Comment:
Why do we have hardcoded and loaded-from-file IMAGE_CLASS? Could we just have one?

Reply:
Hi, do you think it is better to load from a file rather than define it manually? I mean in this case, I'll load `classes.txt` directly in line 49 and to do that I need to copy the file locally into `cub_bahasa` folder.

Reply:
The latest commit removed the loaded-from-file IMAGE_CLASS

Reply:
Personally, I prefer it to be loaded from the file. However, the other data loaders here seem to hard code labels so I guess it is the norm. Thanks for updating!